src/workerd/api/sockets.h

```diff
@@ -46,6 +47,11 @@ class Socket: public jsg::Object {
   jsg::Promise<void> close(jsg::Lock& js);
   // Closes the socket connection.
 
+  jsg::Ref<Socket> startTls(jsg::Lock& js);
+  // Flushes write buffers then performs a TLS handshake on the current Socket connection.
+  // The current `Socket` instance is closed. All new operations should be performed on the new
+  // `Socket` instance.
```

Comment:
Nit: I'd highlight here also what happens to the original Socket's `readable` and `writable` instances.